src/Mirza/BusinessRegistry/GenerateUtils.hs

```diff
@@ -25,7 +25,8 @@ dummyBusiness unique = do
 dummyUser :: T.Text -> GS1CompanyPrefix -> IO NewUser
 dummyUser unique business_uid = do
   passwordEntropy <- randomText
-  let newUserEmailAddress = unsafeEmailAddress (encodeUtf8 unique) "example.com"
+  randomEmail <- randomText
```

Comment:
The problem is when the function is used to generate 1 user at a time (so 1 user will be generated each time, and it will be called multiple times). For example: https://github.com/data61/Mirza/blob/master/src/Mirza/SupplyChain/PopulateUtils.hs#L123

That means that the user `test1@example.com` will be produced every time, and therefore will clash.
Having a random text takes care of the problem. I thought that was the quickest solution to this problem.
Another solution to this would be to generate a random number and attach it to the end of the username of the email, but that's much more likely to clash, and it still isn't deterministic.
If anything, since these are dummy users, we can look at the database and pick a user if need be (as email addresses are visible).
If there is another solution that you can think of, I am more than happy to adopt that instead :)

Reply:
So I think I know how the problem started (and why it wasn't a problem in SCS).
There should have been/was a parameter `[T.Text]` which is a list of unique names that somehow got removed, and what was being used as the email username was `testName <> show n`. I have fixed that now in 7f965379f3994764b76d1f34a9fba9409e99f611. Let me know if that doesn't resolve the issue.